### DISCUSSION TITLE
file input within controller

### DISCUSSION THREAD

**Discussion by daniele-bottelli** [react-hook-form/react-hook-form#9305] [Q&A] [answered]
### Version Number

7.33.0

### Codesandbox/Expo snack

https://codesandbox.io/s/zealous-khorana-u3msmw

### Steps to reproduce

Wrap a file input in a Controller tag
Select a file in the form an press submit button
The selected file value is the file path ( see attachment2 in codesandbox )

This issue do not occur when using register hook ( see attachment1 in codesandbox )

Please do not tell me to use register hook, in my use case it's not the best solution

### Expected behaviour

The selected file value should be a FileList

EDIT: I found a solution according to #5394 . I'd expect Controller and register hook to work in the same way

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@daniele-bottelli 

Component `Controller` or hook `useController` from react-hook-form are built to support cases with custom field element in web and also field element from React Native where accessing to DOM reference might be not available. It cannot be be worked as a same way with method `register`.

Please use the solution that you found in discussion https://github.com/react-hook-form/react-hook-form/discussions/5394 to handle case file upload like this.

**Reply by daniele-bottelli** [reply]
In my opinion if using Controller I have to do
`onChange={(e) => { filed.onChange(e.target.files) }}`
I should do the same while using register hook. It makes api harmonious.
In addition a better documentation would be appreciated :)
